mesa/time.py

```diff
@@ -142,12 +142,12 @@ class SimultaneousActivation(BaseScheduler):
 
     def step(self) -> None:
         """Step all agents, then advance them."""
-        agent_keys = list(self._agents.keys())
+        agent_keys = self._agents.keys()
```

Comment:
I think you can combine with the next line to do:
```python
for agent in self._agents.values():
    agent.step()
```

Reply:
right, it seems unnecessary to step the keys and not the values

Reply:
wait but maybe we can even not calculate everytime the keys!